pandas/tests/tools/test_to_datetime.py

```diff
@@ -2550,11 +2550,49 @@ def test_day_not_in_month_raise(self, cache):
             ):
                 to_datetime("2015-02-29", errors="raise", cache=cache)
 
-    @pytest.mark.parametrize("arg", ["2015-02-29", "2015-02-32", "2015-04-31"])
-    def test_day_not_in_month_raise_value(self, cache, arg):
-        msg = f'time data "{arg}" at position 0 doesn\'t match format "%Y-%m-%d"'
+    @pytest.mark.parametrize(
+        "arg, format, msg",
+        [
+            (
+                "2015-02-29",
+                "%Y-%m-%d",
+                '^time data "2015-02-29" doesn\'t match format "%Y-%m-%d", '
+                "at position 0$",
+            ),
+            (
+                "2015-29-02",
+                "%Y-%d-%m",
+                "^day is out of range for month, at position 0$",
+            ),
+            (
+                "2015-02-32",
+                "%Y-%m-%d",
+                '^time data "2015-02-32" doesn\'t match format "%Y-%m-%d", '
+                "at position 0$",
+            ),
+            (
+                "2015-32-02",
+                "%Y-%d-%m",
+                '^time data "2015-32-02" doesn\'t match format "%Y-%d-%m", '
+                "at position 0$",
+            ),
+            (
+                "2015-04-31",
+                "%Y-%m-%d",
+                '^time data "2015-04-31" doesn\'t match format "%Y-%m-%d", '
+                "at position 0$",
+            ),
+            (
+                "2015-31-04",
+                "%Y-%d-%m",
+                "^day is out of range for month, at position 0$",
+            ),
```

Comment:
note how the ISO and non-ISO messages don't quite match yet - this PR just ensures that they both show the position of the error

https://github.com/pandas-dev/pandas/pull/50242 will ensure that the messages match